data_structures/linked_list/__init__.py

```diff
@@ -41,6 +41,17 @@ def add(self, item: Any, position: int = 0) -> None:
         >>> linked_list.add(4, 2)
         >>> print(linked_list)
         3 --> 2 --> 4 --> 1
+
+        # Test adding to a negative position
+        >>> linked_list.add(5, -3)
+        Traceback (most recent call last):
+            ...
+        ValueError: Position must be non-negative
+
+        # Test adding to an out-of-bounds position
+        >>> linked_list.add(5, 4)
+        >>> print(linked_list)
+        3 --> 2 --> 4 --> 1 --> 5
```

Comment:
This isn't testing an out-of-bounds position, but rather it's testing the end of the list. Since the indices are 0, 1, 2, and 3, adding at index 4 means adding to the end of the list.

We should still keep this test, though. Instead, could you add one more test to actually test an out-of-bounds position? For example:
```py
>>> linked_list.add(0, 6)
```
This test tries to add 0 to the list at index 6, which is definitely beyond the end of the list.

Reply:
out of bounds both positive and negative, got it